server/src/logic/middleware/model-validation.middleware.ts

```diff
@@ -37,4 +37,39 @@ export class ModelValidationMiddleware {
         });
     });
   }
+
+  /**
+   * This validation approache enables also validation of nested objects
+   * @param objToValidate 
+   * @param req 
+   * @param validationGroups 
+   */
+  public validateObject (objToValidate: Validatable, req: Request, validationGroups: ValidationGroup[]): Promise<any> {
```

Comment:
To remove this function we need to adapt the user routes and model to fit the new one.